sympy/geometry/line.py

```diff
@@ -2121,10 +2121,12 @@ def __new__(cls, p1, p2, **kwargs):
         p2 = Point(p2, dim=2)
         if p1 == p2:
             return p1
-        if (p1.x > p2.x) == True:
-            p1, p2 = p2, p1
-        elif (p1.x == p2.x) == True and (p1.y > p2.y) == True:
-            p1, p2 = p2, p1
+        evaluate = kwargs.get('evaluate', False)
```

Comment:
You got the sense wrong, I think: if evaluate=False you shouldn't try to order the points. Would anything break if Segments didn't try to reorder their points? Line doesn't do so. All that might need to be done is the equals method be modified to return `isinstance(o, self.func) and list(ordered(self.args)) == list(ordered(o.args))` instead of `self == o`.